### ISSUE TITLE
Option amnesia

### ISSUE THREAD

**Issue by GoogleCodeExporter** [max99x/pagemon-chrome-ext#61]
```
What steps will reproduce the problem?

Not quite sure. I changed the sorting option to be by last changed, the check 
interval to 1 day and the view all function to "view changes". After a reboot, 
possibly when Chrome was not shut down gracefully (not sure), options were 
partially forgotten.
In particular, the view all function reverted to "view originals" and the check 
interval reverted to 2 hours and 59 mins (I think).

What is the expected output? What do you see instead?

Settings should be saved either immediately (or there should be a "save" 
button".

What version of the Chrome are you using? On what operating system?

6.0.453.1 with Windows XP Pro SP3

Please provide any additional information below.

```

Original issue reported on code.google.com by `gkaemp...@gmail.com` on 4 Jul 2010 at 2:25

**Comment by GoogleCodeExporter** [comment]
```
Unless it is reproducible, this is likely a problem with the dev version of 
Chrome. Marking as invalid, but will reopen if this happens to anyone on stable 
or steps are provided to reliably reproduce it.
```

Original comment by `max99x` on 10 Jul 2010 at 7:46
* Changed state: **Invalid**
* Added labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
This still persists on 6.0.472.14 dev. To reproduce, just remove a page from 
the options and export. The removed page is still in the export. Something just 
doesn't get saved to the DB or so it seems...
```

Original comment by `gkaemp...@gmail.com` on 1 Aug 2010 at 9:56

**Comment by GoogleCodeExporter** [comment]
```
I believe that 6.0.472.55 is now considered the stable version for Chrome and 
this issue still exists (and is quite annoying). Could you possibly check it 
out again? (I would guess that it shouldn't be too hard to reproduce (see 
above) and fix).
Thanks for a great extension!
```

Original comment by `gkaemp...@gmail.com` on 8 Sep 2010 at 8:38

**Comment by GoogleCodeExporter** [comment]
Original comment by `max99x` on 8 Sep 2010 at 1:49
* Changed state: **New**

**Comment by GoogleCodeExporter** [comment]
```
Could not reproduce on 6.0.472.59. May be a case of database corruption. Try 
exporting, deleting everything then reimporting. A reliable (but permanent!) 
way to delete everything is to run the following commands in the console 
(Ctrl+Shift+J in the options page) then restart Chrome or disable/enable PM:

executeSql('DROP TABLE pages');
localStorage.clear();

Note that this way you'll lose global settings (badge color, custom sounds, 
etc.).
```

Original comment by `max99x` on 21 Sep 2010 at 5:13
* Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
I followed your explanation as follows:
Exported the pages I monitor (68 in total).
Went to the PM options page and opened the console (Crtl+Shift+J).
Entered the two commands you mentioned.
restarted Chrome.
enabled/disabled the extension (and in another version uninstalled and 
reinstalled).
Went to the options: No pages being monitored.
Imported the same text I exported.
Got a response: Successfully imported 68 pages
Unfortunately no pages appear in the options page and nothing is being 
monitored.
This totally broke my configuration!
How can I be sure that I really uninstalled the extension? Could there be meta 
data in the exported file that causes corruption? Too many pages?
```

Original comment by `gkaemp...@gmail.com` on 21 Sep 2010 at 10:28

**Comment by GoogleCodeExporter** [comment]
```
Ok, this is strange. To check if there's anything still in the database, after 
deleting and restarting, go to the console, then to the Storage tab. From there 
go to Databases -> pages -> pages and make sure in the right pane it says "The 
"pages" table is empty.".

When importing check if the console contains any error messages. If not, 
refresh the page and see if the problem persists (check for error messages 
again). Also, just in case, make sure you have v3.0.1 (released earlier today) 
installed - this will produce error messages with correct line numbers among 
other things.

Also a good test would be to try importing the exported pages into a clean 
Chrome profile (by running Chrome with a --user-data-dir command line parameter 
that points to an arbitrary temporary folder).

If all this fails, and you have no important private information in the pages 
list, you can send me the exported pages list for debugging (my email is my 
username at gmail).
```

Original comment by `max99x` on 21 Sep 2010 at 10:50

**Comment by GoogleCodeExporter** [comment]
```
"The pages table is empty" is indeed what I see before and after the import.
The import finished without any errors in the console (and I get the "Success" 
message popup).
Yes, I installed V3.01.
I tried running with --user-data-dir=="c:\temp".
The import there failed in a similar way, but reported importing 279 pages 
successfully. No console errors either :(
I think I will send you my file...
```

Original comment by `gkaemp...@gmail.com` on 21 Sep 2010 at 11:31

**Comment by GoogleCodeExporter** [comment]
```
Solved by reinstalling to version 6.0.472.62. All looks fine now.
What I did:
1. saved my "Default" profile from where it was located by copying to another 
directory (this is OS dependent).
2. Uninstalled Chrome and kept the profile info.
3. Installed Chrome from the normal Google link (or via Google Pack if that 
fails). Because I kept my profile info, all extensions where there as before.
4. Imported the bookmarks I had saved.
5. Everything worked again...

I didn't check if it would now also work after an upgrade to the 7.x.x.x 
version...

Thanks for the help!
```

Original comment by `gkaemp...@gmail.com` on 26 Sep 2010 at 4:47

**Comment by GoogleCodeExporter** [comment]
Original comment by `max99x` on 27 Sep 2010 at 5:49
* Changed state: **Done**
